crates/signer/src/manager/dirk.rs

```diff
@@ -0,0 +1,917 @@
+                    let key_name =
+                        dist_account.name.split_once("/").map(|(_, n)| n).unwrap_or_default();
+                    let is_proxy = key_name.contains('/');
+                    trace!(?dist_account.name, "Adding distributed account to hashmap");
+
+                    // Find the participant ID for this host from the participants list
+                    let participant_id = dist_account
+                        .participants
+                        .iter()
+                        .find(|p| p.name == domain)
+                        .map(|p| p.id)
+                        .ok_or_else(|| {
+                            eyre::eyre!(
+                                "Host {} not found in distributed account participants",
+                                domain
+                            )
+                        })?;
+
+                    accounts
+                        .entry(hex::encode(public_key))
+                        .and_modify(|account| {
+                            if !account.hosts.iter().any(|host| host.domain == domain) {
+                                account
+                                    .hosts
+                                    .push(HostInfo { domain: domain.clone(), participant_id });
+                            }
+                        })
+                        .or_insert_with(|| Account {
+                            wallet: wallet.to_string(),
+                            name: key_name.to_string(),
+                            public_key: Some(public_key),
+                            hosts: vec![HostInfo { domain: domain.clone(), participant_id }],
+                            wallet_type: WalletType::Distributed,
+                            signing_threshold: dist_account.signing_threshold,
+                            is_proxy,
+                        });
+                }
+            }
+        }
+
+        Ok(Self {
+            chain,
+            channels,
+            accounts,
+            unlock: config.unlock,
+            secrets_path: config.secrets_path,
+            proxy_store: None,
+        })
+    }
+
+    fn accounts(&self) -> Vec<Account> {
+        self.accounts.values().cloned().collect()
+    }
+
+    fn accounts_non_proxy(&self) -> Vec<Account> {
+        self.accounts.values().filter(|a| !a.is_proxy).cloned().collect()
+    }
+
+    pub fn with_proxy_store(self, proxy_store: ProxyStore) -> eyre::Result<Self> {
+        Ok(Self { proxy_store: Some(proxy_store), ..self })
+    }
+
+    /// Get all available accounts in the `self.accounts` wallets
+    pub async fn get_all_accounts(&self) -> Result<Vec<DirkAccount>, SignerModuleError> {
+        let mut seen_accounts: HashSet<Vec<u8>> = HashSet::new();
+        let mut unique_accounts = Vec::new();
+
+        // Query all channels and combine results
+        for channel in self.channels.values() {
```

Comment:
Since now we have a HashMap with all accounts and will only use the ones in the config file, can we just use the HashMap?

Reply:
You mean just do this logic once at startup and then return the hashmap? My question is if the accounts in the Dirk host can't change whilst the program is running, if not I can simplify this probably yes.

Reply:
Should be solved in 5d636181bcd8cac521154280e642710e493bb4a3 and 93821c3f41e5d8962615103abf6379315e9ae7b4